sim/dex.ts

```diff
@@ -815,6 +815,16 @@ export class ModdedDex {
 				} else if (ruleSpec[0] === 'complexBan') {
 					const complexBan: Data.ComplexBan = ruleSpec.slice(1) as Data.ComplexBan;
 					ruleTable.addComplexBan(complexBan[0], complexBan[1], complexBan[2], complexBan[3]);
+				} else if (ruleSpec[0] === 'teamlength') {
+					const onValidate = ruleSpec[2] as number;
```

Comment:
You shouldn't need to cast this; you should just define `ruleSpec` as a union type such that this will automatically be true.

Reply:
Akasianse was looking for help on their PR and mentioned this comment. `ruleSpec[2]` is already the union type `string | number | number[]`, due to the return type of `validateRule`. It is forced to be a `number` by `rule.teamLength.validate : [number, number]`. There is no way to avoid the cast without modifying `validateRule` to return a discriminated union with tags `'complexBan'`, `'teamlength'`, `'level'`, etc. I think that's a good idea but I said that they should talk to you first before modifying functions outside the scope of their PR.

Reply:
Yes, the discriminated union was indeed what I was talking about.

Reply:
Also, `validateRule`'s return signature is already modified in this PR.